subsys/bluetooth/mesh/crypto.c

```diff
@@ -68,7 +76,7 @@ int bt_mesh_k2(const uint8_t n[16], const uint8_t *p, size_t p_len,
 		return err;
 	}
 
-	err = bt_mesh_aes_cmac_one(salt, n, 16, t);
+	err = bt_mesh_aes_cmac_one_raw_key(salt, n, 16, t);
```

Comment:
Why is not salt/t/out defined as bt_mesh_raw_key_t? As that is what bt_mesh_aes_cmac_one_raw_key takes as input. There are more of these in this file.

Reply:
salt is not key. This is separate entity according to encryption algorithms